pkg/controller/install/openshift/openshift.go

```diff
@@ -86,6 +87,168 @@ func EnsureOpenshiftIngress(c client.Client, scheme *runtime.Scheme, instance *s
 	return nil
 }
 
+// EnsureConfigMapForCrt ensures crt injection into the configmap
+func EnsureConfigMapForCrt(c client.Client, scheme *runtime.Scheme, instance *servingv1alpha1.Install) error {
+	namespace := instance.GetNamespace()
+	if routeExists, err := kindExists(c, "route", "route.openshift.io/v1", namespace); err != nil {
+		return err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return nil
+	}
+
+	if err := updateConfigMapForCrt(c, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateConfigMapForCrt(c client.Client, namespace string) error {
```

Comment:
Once you rebase, you'll see that this "preInstall" function's signature should just expect an Install instance, from which you can obtain the namespace. And there's a package var called `api` you can use for your client calls.

Reply:
Yes..I have changed according to the latest code.